### ISSUE TITLE
Consider Lzip (or Zstd) for tarball compression

### ISSUE THREAD

**Issue by toonn** [fish-shell/fish-shell#7761]
Since the switch to xz is recent and I just came across [this reasonable argument against it](http://www.nongnu.org/lzip/xz_inadequate.html) by the author of Lzip, I figured I'd bring it up.

Lzip is supported by both GNU and BSD Tar so it's a potential candidate. I'm surprised Zstd wasn't considered, are there backwards compatibility considerations at play?

**Comment by faho** [comment]
> this reasonable argument against it by the author of Lzip

It's... quite unreasonable, and always has been. We don't care about xz being a "container format" or "extensible" or "fragmented by design", as the xz files we produce work, and work basically everywhere.

We don't particularly care about xz protecting any fields or any form of error detection, we have sha256 for that.

We don't care about theoretical efficiency, the benchmarks work out well enough.

It simply doesn't apply. We're happy with xz, so I'm closing this.

**Comment by toonn** [comment]
Fair enough. Any reason Zstd wasn't considered? The original motivation to move to xz seemed to be compression ratio and Zstd is impressive in that regard.

**Comment by faho** [comment]
Not supported enough yet - we also support RHEL and such.

**Comment by zanchey** [comment]
Debian and Fedora are both using xz, and interestingly Fedora has rejected Zstd (at least for now) as its main packaging compression format.

See #5460.

**Comment by toonn** [comment]
@zanchey, yes, though it being popular isn't actually a good reason to use it, by popularity everyone should be using plain zips. The article I linked mentions debian choosing xz, and links to a discussion on their mailing liist. [Guix opted for Lzip](https://guix.gnu.org/blog/2019/substitutes-are-now-available-as-lzip/) a couple years ago.

The interesting arguments imo are wrt to long-term archiving. Lzip and Xz both perform similarly because they're both based on LZMA, but the Lzip format has far fewer shortcomings in the face of file corruption. Fish is an interesting enough project that I think digitally preserving it should be a consideration.

Since this is about tarballs and not plain compressed files, whether or not distributions ship xz or lzip isn't really important, since both GNU and BSD Tars support both formats just fine. GNU Tar supports Xz and Lzip since version 1.22 and 1.23, respectively, LibArchive/BSD Tar since 2.7.0 and 3.0.0a, respectively. So both formats have been supported for a decade or more.

**Comment by zanchey** [comment]
Being popular and well-understood within the ecosystem is an extremely important aspect of the choice.

Zstd produces a marginally-larger compressed file (even at compression level 19). lzip is basically comparable at `-6` and `-9`, which you'd expect.

I'll leave things with xz for now, but if we need to revisit in the future.
